Helper/Cache.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace EMS\CommonBundle\Helper;
+
+use Symfony\Component\HttpFoundation\Response;
+
+class Cache
+{
+    /** @var string */
+    private $hashAlgo;
+
+    public function __construct(string $hashAlgo)
+    {
+        $this->hashAlgo = $hashAlgo;
+    }
+
+    public function generateEtagFromResponse(Response $response, ?\DateTime $lastUpdateDate, bool $immutableRoute): void
```

Comment:
This function is generating an etag, and making the response cacheable, that is invalid against the S from SOLID (single responsability).

If you have an etag, you call makeResponseCacheable directly. If you do not have an etag, you create one with this function (should return an etag, not void), and then you call makeResponse with your etag.